src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -915,7 +915,8 @@ public async Task CopyFromSourceToDestinationAsync()
                         {
                             await _destination.WriteAsync(new ReadOnlyMemory<byte>(_arrayPoolBuffer, 0, bytesRead), _cancellationToken).ConfigureAwait(false);
                         }
-                        else
+
+                        if (_deflateStream._inflater.NeedsInput() || _deflateStream._inflater.Finished())
```

Comment:
Is there any benefit to checking this before the first Inflate call?  If so, you could just make it the while condition, e.g.
```C#
while (!_deflateStream._inflater.NeedsInpu() && !_deflateStream._inflator.Finished())
```

Reply:
I couldn't do this because NeedsInput is not a sufficient boundary condition.  See my other comment.  I was able to move finished which I think reads better, behaves the same way, and achieves the same perf (it's merely a field check).